R/plot.R

```diff
@@ -190,24 +191,35 @@ plot.incidence <- function(x, ..., fit = NULL, stack = is.null(fit),
   ## counts the number of days within each interval.
 
   ## Adding a variable for width in ggplot
-  df$interval.days <- get_interval(x, integer = TRUE)
+  df$interval_days <- get_interval(x, integer = TRUE)
   ## if the date type is POSIXct, then the interval is actually interval seconds
   ## and needs to be converted to days
   if (inherits(df$dates, "POSIXct")) {
-    df$interval.days <- df$interval.days * 86400 # 24h * 60m * 60s
+    df$interval_days <- df$interval_days * 86400 # 24h * 60m * 60s
   }
   ## Important note: it seems safest to specify the aes() as part of the geom,
   ## not in ggplot(), as it interacts badly with some other geoms like
   ## geom_ribbon - used e.g. in projections::add_projections().
-  x_axis <- "dates + (interval.days/2)"
+
+
+  ## add mid-interval positions for x-axis
+
+  ## THIS BREAKS THE PLOT WITH ggplot2_3.3.0
+  ## See bug reports at:
+  ## https://github.com/reconhub/incidence/issues/119
+  ## https://github.com/tidyverse/ggplot2/issues/3873
+  ## Temporary fix: changing placement to default of ggplot2::scale_x_date
+  
+  df$x_dates <- df$dates + (df$interval_days / 2)
```

Comment:
x_dates is unused here. Suggest to remove this line and the dead code below.

Reply:
Yup, my bad, removing.